core/domain/improvements_services.py

```diff
@@ -212,3 +213,27 @@ def apply_changes_to_model(task_entry, task_entry_model):
         task_entry_model.resolved_on = task_entry.resolved_on
         changes_were_made_to_model = True
     return changes_were_made_to_model
+
+
+def update_task_dict_with_resolver_settings(task_entry):
+    """Makes changes to the given task dict according to resolver settings.
```

Comment:
This is not really what it does, It returns a task entry dict where username and profile pictures is inserted. Also I think this can be moved to _controllers/improvements.py_

Reply:
@vojtechjelinek  I will update the name and docstring. I have a query. Can we use modules from core.controllers in domain files? Because that would be required if this function is moved to _controllers/improvements.py_.

Reply:
No we cannot. But I don't understand why would need to import anything from it in domain.

Reply:
The _core/domain/improvement_services_test.py_ uses this function in some of the tests. That is why I defined the function in _core/domain/improvements_services.py_.

Reply:
@vojtechjelinek Posting the references to the usages in _core/domain/improvement_services_test.py_.
https://github.com/oppia/oppia/blob/ac3e78723733c748ef46266a8d1faba8d3e8fde8/core/domain/improvements_services_test.py#L181
https://github.com/oppia/oppia/blob/ac3e78723733c748ef46266a8d1faba8d3e8fde8/core/domain/improvements_services_test.py#L293
https://github.com/oppia/oppia/blob/ac3e78723733c748ef46266a8d1faba8d3e8fde8/core/domain/improvements_services_test.py#L362
https://github.com/oppia/oppia/blob/ac3e78723733c748ef46266a8d1faba8d3e8fde8/core/domain/improvements_services_test.py#L387

The calls for _toDict()_ here are basically the ones which would be replaced by  `update_task_dict_with_resolver_settings`

Reply:
Done.